src/main/java/org/everit/osgi/ecm/annotation/metadatabuilder/MetadataBuilder.java

```diff
@@ -108,8 +108,41 @@
  */
 public final class MetadataBuilder<C> {
 
+  /**
+   * Compares {@link AttributeMetadata} classes based on their priority and than their names.
+   */
+  private static final class AttributeMetadataComparator
+      implements Comparator<AttributeMetadata<?>>, Serializable {
+
+    private static final long serialVersionUID = -7796104393729198249L;
+
+    @Override
+    public int compare(final AttributeMetadata<?> attr1, final AttributeMetadata<?> attr2) {
+      if (attr1 == null) {
+        return -1;
```

Comment:
why should we return with -1 if attr1 is null? If both are null, they are still equal.